src/lib/chips/chip-list.ts

```diff
@@ -85,14 +153,153 @@ export class MdChipList implements AfterContentInit, OnDestroy {
 
-  constructor(protected _renderer: Renderer2, protected _elementRef: ElementRef,
-              @Optional() private _dir: Directionality) {
+  _selectionModel: SelectionModel<MdChip>;
+
+  /** Comparison function to specify which option is displayed. Defaults to object equality. */
+  private _compareWith = (o1: any, o2: any) => o1 === o2;
+
+  get selected(): MdChip[] {
+    return this._selectionModel.selected;
+  }
+
+  /**
+   * A function to compare the option values with the selected values. The first argument
+   * is a value from an option. The second is a value from the selection. A boolean
+   * should be returned.
+   */
+  @Input()
+  get compareWith() { return this._compareWith; }
+  set compareWith(fn: (o1: any, o2: any) => boolean) {
+    this._compareWith = fn;
+    if (this._selectionModel) {
+      // A different comparator means the selection could change.
+      this._initializeSelection();
+    }
+  }
+
+  /** Required for FormFieldControl */
+  @Input()
+  get value() { return this._value; }
+  set value(newValue: any) {
+    this.writeValue(newValue);
+    this._value = newValue;
+  }
+
+  /** Required for FormFieldControl. The ID of the chip list */
+  @Input()
+  set id(value: string) {
+    this._id = value;
+    this.stateChanges.next();
+  }
+  get id() { return this._id || this._uid; }
+
+  /** Required for FormFieldControl. Whether the chip list is required. */
+  @Input()
+  set required(value: any) {
+    this._required = coerceBooleanProperty(value);
+    this.stateChanges.next();
+  }
+  get required() {
+    return this._required;
+  }
+
+  /** For FormFieldControl. Use chip input's placholder if there's a chip input */
+  @Input()
+  set placeholder(value: string) {
+    this._placeholder = value;
+    this.stateChanges.next();
+  }
+  get placeholder() {
+    return this._chipInput ? this._chipInput.placeholder : this._placeholder;
+  }
+
+  /** For FormFieldControl. If any of the chips has focus, or the chip input has focus */
+  get focused() {
+    return this.chips.some(chip => chip._hasFocus) ||
+      (this._chipInput && this._chipInput.focused);
+  }
+
+  /** For FormFieldControl. The chip list is empty if there's no chip and there's no input */
+  get empty(): boolean {
+    return (!this._chipInput || this._chipInput.empty) && this.chips.length === 0;
+  }
+
+  /** For FormFieldControl. The disabled is not depend on chip input */
+  @Input()
+  get disabled() { return this.ngControl ? this.ngControl.disabled : this._disabled; }
+  set disabled(value: any) { this._disabled = coerceBooleanProperty(value); }
+
+  get errorState(): boolean {
+    const isInvalid = this.ngControl && this.ngControl.invalid;
+    const isTouched = this.ngControl && this.ngControl.touched;
+    const isSubmitted = (this._parentFormGroup && this._parentFormGroup.submitted) ||
+      (this._parentForm && this._parentForm.submitted);
+
+    return !!(isInvalid && (isTouched || isSubmitted));
```

Comment:
This looks pretty similar to the select's default error state here: https://github.com/angular/material2/blob/master/src/lib/select/select.ts#L678
 
Could we share it somewhere? There is a default errorStateMatcher - could we update it to add the control existence checks?

Reply:
We have #6741 to track sharing common code.

Reply:
Ah, good to know, thanks.